### PR TITLE
Second mild refactoring attempt of Setupper

### PR DESCRIPTION
**Pull request by EdwardKaravakis**
Second mild refactoring attempt of Setupper / Setupper ATLAS Plugin to address most of the comments received after the meeting with Tadashi and Fernando
- Removing obsolete code
- Correcting a bit the docs

This PR does **not** include splitting of the setup_source and setup_destination methods of the ATLAS plugin.

Your code has been rated at 9.13/10 (previously the first mild refactored code was 8.73/10)

### CODE REVIEW

File: pandaserver/brokerage/broker.py [hunk 1]
```diff
@@ -243,30 +243,30 @@ def sendMsgToLoggerHTTP(msgList, job):
 
 
 # get T2 candidates when files are missing at T2
-def getT2CandList(tmpJob, siteMapper, t2FilesMap):
```
Comment (fbarreir) [inline pandaserver/brokerage/broker.py:246]: I think we concluded in the discussion that T1s-> nuclei and T2s->satellites? With your changes we are entering a mixed nomenclature mode using both together.

File: pandaserver/dataservice/setupper_atlas_plugin.py [hunk 2]
```diff
@@ -96,93 +97,93 @@ def run(self) -> None:
...
+                            self.setup_destination(start_idx=tmp_index_job, n_jobs_in_loop=i_bunch)
+                            tmp_index_job += i_bunch
+                            i_bunch = 0
+                    # increment
+                    i_bunch += n_jobs_per_dis
+                # remaining
+                if i_bunch != 0:
+                    self.setup_destination(start_idx=tmp_index_job, n_jobs_in_loop=i_bunch)
+            else:
+                # make one sub per job so that each job doesn't have to wait for others to be done
```
Comment (fbarreir) [inline pandaserver/dataservice/setupper_atlas_plugin.py:172]: sub -> sub dataset

File: pandaserver/dataservice/setupper_atlas_plugin.py [hunk 3]
```diff
@@ -59,14 +60,14 @@ def __init__(self, taskBuffer, jobs: List, logger, **params: Dict) -> None:
         self.all_replica_map = {}
         # replica map for special brokerage
         self.replica_map_for_broker = {}
-        # available files at T2
-        self.available_lfns_in_t2 = {}
+        # available files at satellite sites
+        self.available_lfns_in_satellites= {}
```
Comment (fbarreir) [inline pandaserver/dataservice/setupper_atlas_plugin.py:64]: self.available_lfns_in_satellites _space_= {}

File: pandaserver/dataservice/setupper_atlas_plugin.py [hunk 4]
```diff
@@ -847,7 +846,7 @@ def subscribe_dispatch_data_block(self) -> None:
                 # destination
                 tmp_dst_id = job.computingSite
                 tmp_site_spec = self.site_mapper.getSite(job.computingSite)
-                scope_tmp_site_input, _ = select_scope(tmp_site_spec, job.prodSourceLabel, job.job_label)
+                scope_tmp_site_input, scope_tmp_site_output = select_scope(tmp_site_spec, job.prodSourceLabel, job.job_label)
```
Comment (fbarreir) [inline pandaserver/dataservice/setupper_atlas_plugin.py:849]: when a variable is not used later, but you need a placeholder for the return of a function, it's common to use "_" to throw some garbage in it. I would not replace these, unless you will use scope_tmp_site_output later.

**Review by fbarreir — COMMENTED** [review]
Generally good and great to get rid of some of the conditions that were duplicating parts of the code. 
Just that now there is a mixed T1/T2 and nucleus/satellite nomenclature - this would be better to solve in a future round. 
Some changes are sensible, this needs to be tested and run for a good period on one machine only.

File: pandaserver/brokerage/broker.py (hunk 1 above)
Comment (EdwardKaravakis) [reply pandaserver/brokerage/broker.py:246]: Will replace it, it's just that it was on broker.py and not on the dataservice folder. I guess you prefer to have it corrected also there, I will update my PR

File: pandaserver/dataservice/setupper_atlas_plugin.py (hunk 4 above)
Comment (EdwardKaravakis) [reply pandaserver/dataservice/setupper_atlas_plugin.py:849]: It is used later on, I had a bug apparently in my previous refactored code

**Comment by EdwardKaravakis** [comment]
Updated PR, do I have the green light to install it on panda server 00 and leave it running there?

**Pull request merged**